plugins-compat-tester-model/src/main/java/org/jenkins/tools/test/model/hook/PluginCompatTesterHook.java

```diff
@@ -47,4 +49,11 @@ default boolean check(Map<String, Object> info) {
       * Check the object used for this hook.
       */
      void validate(Map<String, Object> toCheck);
+     
+     /**
+      * Provides a list of test types executed under the hook (i.e. hooks used to launch other tests rather than surefire.
+      */
+     default Collection<String> getTestTypes() {
```

Comment:
This is a particular need for some plugins, not part (IMO) of any general build cycle process hook mechanism. I do not believe is a good idea to start adding particular methods to cover particular plugins needs to the base interface of the hook mechanism which is aimed to provide a way to override PCT logic for certain plugins. As such it has to be generic, particular needs can be implemented in each individual hook.  

For example, what is the purpose of this method in a `precompilehook` or for a `precheckout` one?

Reply:
Added description inside PR and applied your proposal. Thanks a lot! It improved the previous solution design a lot and simplify implementation. Kudos!